libs/domain/cart/summary/src/summary.component.ts

```diff
@@ -1,40 +1,39 @@
-import { CartComponentMixin, CartController } from '@spryker-oryx/cart';
+import { CartComponentMixin } from '@spryker-oryx/cart';
 import { resolve } from '@spryker-oryx/di';
-import { ContentController } from '@spryker-oryx/experience';
+import { ContentMixin, defaultOptions } from '@spryker-oryx/experience';
 import { SemanticLinkService, SemanticLinkType } from '@spryker-oryx/site';
 import { HeadingTag } from '@spryker-oryx/ui/heading';
-import { asyncValue, hydratable, i18n } from '@spryker-oryx/utilities';
-import { html, TemplateResult } from 'lit';
-import { combineLatest, map } from 'rxjs';
+import {
+  asyncState,
+  hydratable,
+  i18n,
+  valueType,
+} from '@spryker-oryx/utilities';
+import { html, LitElement, TemplateResult } from 'lit';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
 import { CartSummaryOptions } from './summary.model';
 import { styles } from './summary.styles';
 
+@defaultOptions({ maxVisibleQuantity: 99 })
 @hydratable('window:load')
-export class CartSummaryComponent extends CartComponentMixin<CartSummaryOptions>() {
+export class CartSummaryComponent extends CartComponentMixin(
+  ContentMixin<CartSummaryOptions>(LitElement)
+) {
   static styles = styles;
 
-  protected contentController = new ContentController(this);
-  protected cartController = new CartController(this);
-
-  protected link = resolve(SemanticLinkService).get({
-    type: SemanticLinkType.Cart,
-  });
-
-  protected quantity$ = combineLatest([
-    this.cartController.getTotalQuantity(),
-    this.contentController.getOptions(),
-  ]).pipe(
-    map(([quantity, options]) => this.populateQuantity(quantity, options))
+  @asyncState()
+  protected link = valueType(
+    resolve(SemanticLinkService).get({
```

Comment:
resolving inside async state looks like too much IMO. Opinionated ofc, but I like to make it clean, which services compoennt uses